common/src/main/java/feast/common/interceptors/GrpcExceptionInterceptor.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright 2018-2019 The Feast Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.common.interceptors;
+
+import feast.common.logging.AuditLogger;
+import feast.common.logging.LogEvent;
+import io.grpc.ForwardingServerCallListener.SimpleForwardingServerCallListener;
+import io.grpc.Metadata;
+import io.grpc.ServerCall;
+import io.grpc.ServerCall.Listener;
+import io.grpc.ServerCallHandler;
+import io.grpc.ServerInterceptor;
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
+import java.util.NoSuchElementException;
+import org.apache.logging.log4j.Level;
+import org.springframework.security.core.AuthenticationException;
+
+/**
+ * ExceptionInterceptor intercepts exceptions throw when handling GRPC messages. Intercepts uncaught
+ * exceptions and logs to audit logger and returns the appropriate Grpc statuses.
+ */
+public class GrpcExceptionInterceptor implements ServerInterceptor {
+  private AuditLogger auditLog;
+
+  public GrpcExceptionInterceptor(AuditLogger auditLog) {
+    this.auditLog = auditLog;
+  }
+
+  @Override
+  public <ReqT, RespT> Listener<ReqT> interceptCall(
+      ServerCall<ReqT, RespT> call, Metadata headers, ServerCallHandler<ReqT, RespT> next) {
+
+    // Register listener to intercept uncaught exceptions
+    ServerCall.Listener<ReqT> listener = next.startCall(call, headers);
+    return new SimpleForwardingServerCallListener<ReqT>(listener) {
+      @Override
+      public void onHalfClose() {
+        try {
+          super.onHalfClose();
+        } catch (AuthenticationException e) {
+          // TODO: figure out of catching exception breaks spring security
```

Comment:
@woop Does the current E2E auth tests verify that there are no false positives for authentication? (ie should fail to authenticate due to wrong credentials. )

Reply:
#892 These tests will fail your authentication if you try to sign in with a JWT that cannot be verified.

Reply:
No longer relevant  as this was deemed out of scope for this PR.